eureka-client/src/main/java/com/netflix/eureka2/client/channel/RegistrationChannelImpl.java

```diff
@@ -43,17 +40,17 @@ public RegistrationChannelImpl(TransportClient transportClient, RegistrationChan
 
     @Override
     public Observable<Void> register(final InstanceInfo instanceInfo) {
-        if (!moveToState(STATES.Idle, STATES.Registered)) {// State check. Only register if the state is Idle.
+        if (!moveToState(STATES.Idle, STATES.Registered) &&
+            !moveToState(STATES.Registered, STATES.Registered)) {
```

Comment:
We can replace moveToState(STATES.Registered, STATES.Registered) with state.get() == STATES.Registered.